clusterloader2/pkg/provider/aks.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+type AKSProvider struct {
+	features Features
+}
+
+func NewAKSProvider(configs map[string]string) *AKSProvider {
+	return &AKSProvider{
+		features: Features{
+			SupportProbe:                        true,
+			SupportImagePreload:                 true,
+			SupportGrabMetricsFromKubelets:      true,
+			ShouldPrometheusScrapeApiserverOnly: true,
+		},
+	}
+}
+
+func (p *AKSProvider) Name() string {
+	return AKSName
+}
+
+func (p *AKSProvider) Features() *Features {
+	return &p.features
+}
+
+func (p *AKSProvider) GetComponentProtocolAndPort(componentName string) (string, int, error) {
+	return getComponentProtocolAndPort(componentName)
+}
+
+func (p *AKSProvider) GetConfig() Config {
+	return Config{}
+}
+
+func (p *AKSProvider) RunSSHCommand(cmd, host string) (stdout string, stderr string, exit int, err error) {
```

Comment:
nit: let remove names for output params, i.e.
```
func (p *AKSProvider) RunSSHCommand(cmd, host string) (string, string, int error) {
```

Reply:
Same in all other providers

Reply:
ok can do.